### PR TITLE
revamp tutorial to spotify clone

### PR DESCRIPTION
**Pull request by srirachy**
## Changes
1. Add routes to user and playlist data
2. Add code for Spotify Theme, Spotify Content, Sidebar, Navbar, and Playlist components
3. Add styled component files for each component.
4. Move types from components to global type file. Mostly the types in the redux slices are not migrated yet because I'm not sure if the slices are completely finished yet.
5. Add one test for each component so far for TDD practices.

## Purpose
The purpose of the Spotify Theme component is to house the components for Spotify Content, Sidebar, Navbar, and Playlist. Spotify Content holds data for playlist songs and allows user to interact with songs. Sidebar allows user to interact with playlist. Playlist gets the initial playlist data. Navbar houses user info and music search.

## Approach
The initial tutorial found on the Spotify Developer page was definitely a good starting point, but I want to have a more fleshed out music player. Transition to a Spotify Clone is definitely a step in the right direction.

## Learning
Creating and developing a Spotify clones helps me understand how to interact with the Spotify API a lot more, especially when creating requests to the available endpoints. The goal is to eventually have three different themes where Spotify Theme is the learning phase, the second Discover Theme is the actual MVP, and Visualization Theme is an additional feature.

P.S - My bad this push was 800+ lines. I'm almost to the actual feature that this branch is suppose to be. A little disorganized, but it should improve after the music player control buttons are in.

### CODE REVIEW

File: client/src/components/MusicPlayer/MusicPlayer.tsx [hunk 1]
```diff
@@ -8,6 +11,24 @@ import {
 } from '../../styles/MusicPlayerStyle';
 
 function MusicPlayer() {
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
+    const getUserInfo = async () => {
+      const response = await fetch('/auth/me');
```
Comment (javiguerra777) [inline client/src/components/MusicPlayer/MusicPlayer.tsx:18]: This works, but would this still work if you used axios instead? I saw you installed axios for your project.

Comment (srirachy) [reply client/src/components/MusicPlayer/MusicPlayer.tsx:18]: I mostly installed axios to test something. I'm not going to be using axios front-end for this project.

File: client/src/utils/useUserInfo.tsx [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { shallowEqual } from 'react-redux';
+import { useAppSelector } from 'src/store/hooks';
+
+function useUserInfo() {
+  return useAppSelector((state) => state.user, shallowEqual);
+}
+
```
Comment (javiguerra777) [inline client/src/utils/useUserInfo.tsx:7]: nice job setting up redux types in your app for useSelector

File: client/src/utils/Functions.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+export const convertMsToStandardTime = (ms: number) => {
+  const minutes = Math.floor(ms / 60000);
+  const seconds = ((ms % 60000) / 1000).toFixed(0);
+  const addZeroMaybe = Number(seconds) < 10 ? '0' : '';
+  return `${minutes}:${addZeroMaybe}${seconds}`;
+};
```
Comment (javiguerra777) [inline client/src/utils/Functions.ts:6]: I like how you wrote this function to convert Time, good job

File: client/src/types/index.ts [hunk 4]
```diff
@@ -1,7 +1,54 @@
...
+};
+
+export type TrackType = {
+  track: TrackProps;
+};
+
+export type ArtistProp = {
+  name: string;
+};
+
```
Comment (javiguerra777) [inline client/src/types/index.ts:51]: I like how you made a file for your app types

File: client/src/styles/SpotifyThemeStyle.ts [hunk 5]
```diff
@@ -15,9 +15,16 @@ export const SidebarWrapper = styled.div`
   background: linear-gradient(transparent, rgba(0, 0, 0, 1));
   background-color: rgb(32, 87, 100);
 `;
-export const SearchWrapper = styled.div`
+export const NavBarWrapper = styled.div`
   height: 100%;
   width: 100%;
   overflow: auto;
+  &::-webkit-scrollbar {
```
Comment (javiguerra777) [inline client/src/styles/SpotifyThemeStyle.ts:22]: Great idea to style the scroll bar it really improves the UI

File: client/src/components/SpotifyContent/SpotifyContent.tsx [hunk 6]
```diff
@@ -1,5 +1,154 @@
...
+          ),
+          image: track.album.images[2].url,
+          duration: track.duration_ms,
+          album: track.album.name,
+          context_uri: track.album.uri,
+          track_number: track.track_number,
+        })),
+      };
+      if (songData) {
+        dispatch(setPlaylistSongs(songData)); // set song data from selected playlist
```
Comment (javiguerra777) [inline client/src/components/SpotifyContent/SpotifyContent.tsx:56]: since you are using redux, put your api calls into an async thunk

**Review by javiguerra777 — APPROVED** [review]
Incorporate async thunks into your app since you're using redux, and try moving from the fetch api to axios

File: client/src/components/SpotifyContent/SpotifyContent.tsx (hunk 6 above)
Comment (srirachy) [reply client/src/components/SpotifyContent/SpotifyContent.tsx:56]: Thunking is a good idea. But again, I'm not using axios this project. I've been using axios last few projects, so I'm going back to fetch for this one.

**Pull request merged**